util/types/fsp.go

```diff
@@ -43,25 +43,35 @@ func checkFsp(fsp int) (int, error) {
 	return fsp, nil
 }
 
-func parseFrac(s string, fsp int) (int, error) {
+func parseFrac(s string, fsp int) (int, bool, error) {
 	if len(s) == 0 {
-		return 0, nil
+		return 0, false, nil
 	}
 
 	var err error
 	fsp, err = checkFsp(fsp)
 	if err != nil {
-		return 0, errors.Trace(err)
+		return 0, false, errors.Trace(err)
 	}
 
+	// Fill 0 when fsp > string length.
+	if fsp > len(s) {
+		v, err := strconv.ParseInt(s, 10, 64)
+		if err != nil {
+			return 0, false, errors.Trace(err)
+		}
+		return int(float64(v) * math.Pow10(fsp-len(s))), false, nil
+	}
+
+	// Found when fsp <= string length.
```

Comment:
How about this solution like:
```
v, err := strconv.ParseInt(s[:fsp+1], 10, 64)
if err != nil {
    return 0, false, errors.Trace(err)
}
v = (v + 5)/10
```

Then `if fsp > len(s)` should be changed to `if fsp >= len(s)`.